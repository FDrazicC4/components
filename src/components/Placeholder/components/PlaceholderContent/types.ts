import { ReactNode } from 'react';


export interface Props {

    /**
     * Classes to be passed through to the component.
     */
    className?: string,

    /**
     * Any custom content to fill the placeholder that is different than default
     */
    children?: ReactNode,

}
