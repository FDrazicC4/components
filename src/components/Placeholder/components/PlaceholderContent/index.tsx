import React, { ReactElement } from 'react';
import clsx from 'clsx';
import Typography from '../../../Typography';
import { Props } from './types';


function PlaceholderContent({
    className,
    children,
}: Props): ReactElement {


    const placeholderClasses = clsx(
        'brew-Placeholder__content',
        className,
    );


    return (
        <Typography as="p" className={placeholderClasses} variant="body1">
            {children}
        </Typography>
    );


}


export default PlaceholderContent;
