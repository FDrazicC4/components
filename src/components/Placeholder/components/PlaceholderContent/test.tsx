import React from 'react';
import renderer from 'react-test-renderer';
import PlaceholderContent from './index';
import { shallow } from 'enzyme';


describe('PlaceholderContent', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<PlaceholderContent>Foo</PlaceholderContent>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('mounts correctly', () => {
        const placeholderContent = shallow(<PlaceholderContent>Foo</PlaceholderContent>);
        expect(placeholderContent.hasClass('brew-Placeholder__content')).toBe(true);
    });

    it('correctly adds a className', () => {
        const className = `class-${Math.floor(Math.random() * 1000)}`;
        const placeholderContent = shallow(<PlaceholderContent className={className} />);
        expect(placeholderContent.hasClass(className)).toBe(true);
    });

});
