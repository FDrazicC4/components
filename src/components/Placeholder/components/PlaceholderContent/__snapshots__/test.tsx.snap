// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderContent renders correctly 1`] = `
<p
  className="brew-Typography brew-Typography--display-block brew-Typography--variant-body1 brew-Placeholder__content"
>
  Foo
</p>
`;
